### ISSUE TITLE
Unexpected text wrapping behaviour of the RStudio console

### ISSUE THREAD

**Issue by pieterjanvc** [rstudio/rstudio#9967]
### System details

    RStudio Edition : Desktop
    RStudio Version :  2021.09.0 Build 351
    OS Version      :  Windows 10
    R Version       :  4.1.1

### Steps to reproduce the problem
```
#Console width = 89

library(magrittr)

set.seed(10)
sample(c("*", "."), (cli::console_width()+3)*5, 
replace = T) %>% paste0(collapse = "") %>% cat("\n")

sapply(1:5, function(x){
  sample(c("*", "."), (cli::console_width()+3), 
       replace = T) %>% paste0(collapse = "")}) %>% 
  paste(collapse = "\n") %>% cat()
```
![image](https://user-images.githubusercontent.com/52793278/137591706-3c2a5b99-9eb7-4a16-8aa4-5623ab61c066.png)

### Describe the problem in detail
After posting my [question on the RCommunity forum](https://community.rstudio.com/t/unexpected-text-wrapping-behaviour-of-the-rstudio-console/118021), it was suggested to reach out on here.

I'm wondering if anyone knows why the RStudio console is having issues with properly wrapping long strings of text (consisting of various characters). As shown in the example above: using the cli::console_width() function, we can get the "exact" width in characters of the console. Actually, you have to add 3 if you're not counting the default > that's displayed. My console in the example has as width of 89 (89+3) and I'm generating one long string of 5x89 characters and using cat to display it in the console. Then I do the same thing, but I paste 5 strings of 89 together with a linebreak (\n) and then use cat

Note how in the first instance, the wrapping is not correct and is generating incomplete lines with overflow into a 6th line. The number of characters in both is exactly the same, but the wrapping goes wrong. The characters are in a monospace font (you can see that they are aligned even when different) so this should not be the explanation...

I have tested this in other consoles (not RStudio) and this problem does not seem to be happening. Interestingly, if the characters are all the same, the wrapping issue is not present:
![image](https://user-images.githubusercontent.com/52793278/137591800-3e8e775c-e110-4e4e-aa56-37666f5662c7.png)

### Describe the behavior you expected
I expect the text wrapping to make full use of the RStudio console width (as seen in the example with the same characters), and not early and unexpectedly wrap onto the next line, causing messy display of the result. 

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
> The characters are in a monospace font (you can see that they are aligned even when different) so this should not be the explanation...

Can you confirm which font you're using, just in case?

**Comment by pieterjanvc** [comment]
Hello,

I have been taking with others about this issue, and discovered some novel info:
It seems that this issue is only related to RStudio on Windows. All my machines are Windows, but when a colleague tried to reproduce this issue on RStudio MacOS, it did not occur. We then tried it on Linux, and again this issue was not present.

This seems to suggest that it's Windows only, and as I said before, only within the RStudio console, and not any other. I did try it on different Windows machines and was able to reproduce the error.

Hope this helps narrowing it down, and hopefully it can be fixed in a next update.

Kind regards
PJ
